python_modules/libraries/dagster-managed-elements/dagster_managed_elements/cli.py

```diff
@@ -0,0 +1,89 @@
+import importlib.util
+import sys
+from types import ModuleType
+from typing import List
+
+import click
+from dagster_managed_elements.types import ManagedElementDiff, ManagedElementReconciler
+
+MODULE_NAME = "usercode"
+
+
+def load_module(file_path: str) -> ModuleType:
+    """
+    Imports a Python module from a file path.
+    https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly
+    """
+    spec = importlib.util.spec_from_file_location(MODULE_NAME, file_path)
+    if not spec:
+        raise ValueError(f"Could not load module spec from {file_path}")
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[MODULE_NAME] = module
+    loader = spec.loader
+    if not loader:
+        raise ValueError(f"Could not load loader from {file_path}")
+    loader.exec_module(module)
+
+    return module
+
+
+def get_reconcilable_objects(module: ModuleType) -> List[ManagedElementReconciler]:
+    """
+    Collect all ManagedElementReconciler-implementing objects in the root of the
+    module.
+    """
+    return [
+        getattr(module, obj)
+        for obj in dir(module)
+        if isinstance(getattr(module, obj), ManagedElementReconciler)
+    ]
```

Comment:
what led to this set-up? Messing with imports can be spooky so might be nice to avoid if possible. Main spooky elements:
* working directory mattering a lot
* weird corner case behaviors related to how python import system works 

At minimum would be good to layer how things get built up so you don't have to import a whole file for any test or do other shenanigans for complex tests. More sophisticated users may also want to be able to use a cleaner entry point than crawling all module scope things and forcing things be defined at init time.

Reply:
Yeah, this setup is a bit rough at the moment. Originally, the CLI was meant to refer to/operate on a Dagster repository as an entry point, but there were concerns around tying the system to Dagster machinery, so it was suggested to use an adjacent file. This was my attempt to create a very barebones version of the repo loading logic.

I'll make the suggested refactor to make testing easier.

> More sophisticated users may also want to be able to use a cleaner entry point than crawling all module scope things and forcing things be defined at init time.

Do you have any thoughts on a different approach we could use? An entrypoint as sophisticated as the Dagster repository would be great, but is there a better middle-ground for the initial version that we dogfood/test with?

Reply:
I've updated the PR to path to reconcilers a bit differently:

```bash
dagster-managed-elements check -m my_entrypoint_module
```

Users can optionally specify a path to a directory to add to the module load path:

```bash
dagster-managed-elements check -m my-my_entrypoint_module -d ~/Documents/working_directory_containing_module
```

In the vein of uvicorn, they can also specify one or more reconcilers by-attr to check specifically:

```bash
dagster-managed-elements check -m my_entrypoint_module:my_reconcilers.foo,my_reconcilers.bar
```